internal/scraper/nigella.go

```diff
@@ -0,0 +1,49 @@
+package scraper
+
+import (
+	"github.com/PuerkitoBio/goquery"
+	"github.com/reaper47/recipya/internal/models"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func scrapeNigella(root *goquery.Document) (models.RecipeSchema, error) {
+	rs := models.NewRecipeSchema()
+
+	name, _ := root.Find("meta[property='og:title']").Attr("content")
+	before, _, ok := strings.Cut(name, " — ")
+	if ok {
+		name = strings.TrimSpace(before)
+	}
+	rs.Name = name
+
+	yieldNode := root.Find("p[class='serves']")
+	yieldText := strings.Split(yieldNode.Nodes[0].FirstChild.Data, ":")[1]
+	numberPattern, _ := regexp.Compile("[0-9]+")
+	yieldNumMatch := numberPattern.FindString(yieldText)
+	yield, err := strconv.Atoi(yieldNumMatch)
+	if err != nil {
+		return rs, err
+	}
+	rs.Yield.Value = int16(yield)
```

Comment:
This works and the test passes, which is great! There are three ways to extract the yield:
1. Use your approach. When the "[0-9]+" regex will be used elsewhere, it should be precompiled in the [pool of regexes](https://github.com/reaper47/recipya/blob/main/internal/utils/regex/regex.go)
2. Use [findYield](https://github.com/reaper47/recipya/blob/main/internal/scraper/helpers.go#L8) directly on `yieldNode.Text()`.
3. Test [regex.Digit.FindString](https://github.com/reaper47/recipya/blob/main/internal/utils/regex/regex.go#L12) against `yieldNode.Text()`

Any option is fine with me!

Reply:
I didn't realize there was a helper for `findYield`! I'll use that approach, thank you